### ISSUE TITLE
getNodeById() breaks with custom idField

### ISSUE THREAD

**Issue by pixelchutes** [500tech/angular2-tree-component#78]
When using a custom `idField`, the ever so helpful [getNodeById](https://github.com/500tech/angular2-tree-component/blob/5b044b9d3fc62f19c10693204698db8dc6c68849/lib/models/tree.model.ts#L190-L192) method stops working as you might expect. _(It only appears to match on key of: **id**)_

Works when `idField ==  'id'`:
```typescript
let someKey = 123;
tree.treeModel.getNodeById(someKey);
```

But for a custom `idField`, you must rely on something along these lines:
```typescript
let someKey = 123;
tree.treeModel.getNodeBy((node) => node.data[node.treeModel.options.idField] === someKey);
```

It would be great if `getNodeById` could be refactored to work with custom `idField` tree option. 😎

**Comment by adamkleingit** [comment]
Strange, I'm defining a getter `id` that returns the id according to idField,
and using lodash `find` method, passing it an object { id: id }.
I would expect lodash to get the 'id' attribute of each object in the array, thus invoking the getter function.
I wonder why it doesn't work. Any ideas?

Anyway, I will fix in the coming version

**Comment by adamkleingit** [comment]
Update: I just tested it with a custom idField and I don't see the problem you are reporting.
Can you share your code?

Thanks

**Comment by pixelchutes** [comment]
Thanks @adamkleingit

I was dealing with the management of a few different trees able to copy nodes from one to the other, depending on the type of data they were representing. 

The source trees all had a different, custom `idField`, while the destination trees did not. The issue occurred when attempting to "dupe check" against the destination tree, and not factoring in `getNodeById` alone could not suffice given the eventual mixture of node "types".

**TL;DR** I was getting my trees confused 😕 

My mistake. Sorry for the confusion.
